### ISSUE TITLE
[ID] Rename Agencies 

### ISSUE THREAD

**Issue by mhuang-usdr** [usdigitalresponse/usdr-gost#499]
We've heard from a few partners that they're interested in using the tool outside of a strict `Agency` structure. What would be a better name to reflect what our "agencies" are? The term `User Groups` was proposed in the latest meeting with a partner.

Should be part of the domain modeling conversation by @TylerHendrickson

**Comment by mhuang-usdr** [comment]
`Teams` is another one that our partners seem to like

**Comment by mhuang-usdr** [comment]
Waiting on FTE UXD's full redesign before working on this

**Comment by mhuang-usdr** [comment]
Closing as duplicate of #727.
